src/utils.js

```diff
@@ -530,6 +531,19 @@ go.utils = {
             });
     },
 
+    resend_all_subscriptions: function(im, identity) {
+        return go.utils
+            .get_active_subscriptions_by_identity(im, identity.id)
+            .then(function(subscriptions){
+                var promises = [];
+                promises = subscriptions.map(function(result){
+                    var endpoint = 'subscriptions/' + result.id + '/resend';
+                    return promises.push(go.utils.service_api_call('subscriptions', 'post', {}, {}, endpoint, im));
```

Comment:
I'm a little confused by what's happening here.. We're going to end up with `promises` being a list of the results of `.push`, does `.push` even return anything?

I think what we want here is to just return the result of `service_api_call`, `.map` will take care of taking all the return values (which should be promises) and putting them into a list.

So we can just have `var promises = subscriptions.map(function....`, and `return go.utils.service_api_call(....`

Reply:
Thanks for catching this, forgot to change this.